### PR TITLE
Clear PYTHONHOME when using bundled python

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #598

### CODE REVIEW

File: emsdk.py [hunk 1]
```diff
@@ -2575,6 +2575,25 @@ def construct_env(tools_to_activate):
         env_string += 'export ' + key + '="' + value + '";\n'
       else:
         assert False
+      if 'EMSDK_PYTHON' in env_vars:
+        # When using our bundled python we never want the users
```
Comment (kripken) [inline emsdk.py:2579]:
```suggestion
        # When using our bundled python we never want the user's
```

File: emsdk [hunk 2]
```diff
@@ -6,13 +6,31 @@
...
+    # PYTHONHOME or PYTHONPATH
+    # https://github.com/emscripten-core/emsdk/issues/598
+    unset PYTHONHOME
+    unset PYTHONPATH
   fi
 fi
 
+# Look for `python3` first.  This is especially important on macOS (See:
+# https://github.com/emscripten-core/emsdk/pull/273)
+if [ -z "$EMSDK_PYTHON" ]; then
```
Comment (kripken) [inline emsdk:24]: line 10 is identical to line 21, is that intentional?

Comment (sbc100) [reply emsdk:24]: Yes this file is structured as a sequence of attempts to find python.. as soon as one of them passes the other ones are not required.

This matches how we do it in the `emcc` wrapper that calls `emcc.py`

**Review by kripken — APPROVED** [review]
Makes sense, but I don't understand the fine details here...

**Pull request merged**